test/guardian/plug/ensure_not_authenticated_test.exs

```diff
@@ -0,0 +1,115 @@
+      on_failure: {TestHandler, :custom_failure_method}
+    )
+
+    assert handler_opts == {TestHandler, :custom_failure_method}
+  end
+
+  test "init/1 defaults the handler option to Guardian.Plug.ErrorHandler" do
+    %{handler: handler_opts} = EnsureNotAuthenticated.init %{}
+
+    assert handler_opts == {Guardian.Plug.ErrorHandler, :authenticated}
+  end
+
+  test "init/1 with default options" do
+    options = EnsureNotAuthenticated.init %{}
+
+    assert options == %{
+      claims: %{},
+      handler: {Guardian.Plug.ErrorHandler, :authenticated},
+      key: :default
+    }
+  end
+
+  test "init/1 uses all opts as claims except :on_failure, :key and :handler" do
+    %{claims: claims} = EnsureNotAuthenticated.init(
+      on_failure: {TestHandler, :some_method},
+      key: :super_secret,
+      handler: TestHandler,
+      foo: "bar",
+      another: "option"
+    )
+
+    assert claims == %{"foo" => "bar", "another" => "option"}
+  end
+
+  test "it validates claims and calls through if the claims are ok" do
+    claims = %{ "aud" => "token", "sub" => "user1" }
+    conn = :get |> conn("/foo") |> Guardian.Plug.set_claims({ :ok, claims })
+    opts = EnsureNotAuthenticated.init(handler: TestHandler, aud: "token")
+    ensured_conn = EnsureNotAuthenticated.call(conn, opts)
+    assert must_authenticate?(ensured_conn)
+  end
+
+  test "it validates claims and fails if the claims do not match" do
+    claims = %{ "aud" => "oauth", "sub" => "user1" }
+    conn = :get |> conn("/foo") |> Guardian.Plug.set_claims({:ok, claims})
+    opts = EnsureNotAuthenticated.init(handler: TestHandler, aud: "token")
+    ensured_conn = EnsureNotAuthenticated.call(conn, opts)
+    refute must_authenticate?(ensured_conn)
+  end
+
+  test "doesn't call unauthenticated when there's a session with default key" do
+    claims = %{ "aud" => "token", "sub" => "user1" }
+    conn = :get |> conn("/foo") |> Guardian.Plug.set_claims({ :ok, claims })
+    opts = EnsureNotAuthenticated.init(handler: TestHandler)
+    ensured_conn = EnsureNotAuthenticated.call(conn, opts)
+    assert must_authenticate?(ensured_conn)
+  end
+
+  test "doesn't call unauthenticated when theres a session with specific key" do
+    claims = %{ "aud" => "token", "sub" => "user1" }
+    conn = :get
+            |> conn("/foo")
+            |> Guardian.Plug.set_claims({:ok, claims}, :secret)
+    opts = EnsureNotAuthenticated.init(handler: TestHandler, key: :secret)
+    ensured_conn = EnsureNotAuthenticated.call(conn, opts)
+    assert must_authenticate?(ensured_conn)
+  end
+
+  test "calls handler's unauthenticated/2 with no session for default key" do
+    conn = conn(:get, "/foo")
+    opts = EnsureNotAuthenticated.init(handler: TestHandler)
+    ensured_conn = EnsureNotAuthenticated.call(conn, opts)
+    refute must_authenticate?(ensured_conn)
+  end
+
+  test "calls handler's unauthenticated/2 with no session for specific key" do
```

Comment:
I think you might have meant `"calls handler's authenticated/2 with no session for specific key"`? There are a couple more references to `unauthenticated` that need to be changed to `authenticated`.